lib/relaton_ieee/data_fetcher.rb

```diff
@@ -88,27 +88,25 @@ def read_zip(file)
     # @param [String] filename source file
     #
     def fetch_doc(xml, filename) # rubocop:disable Metrics/AbcSize,Metrics/MethodLength,Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
-      doc = Nokogiri::XML(xml).at("/publication")
-      unless doc
+      begin
+        doc = Ieee::Idams::Publication.from_xml(xml)
+      rescue StandardError
         Util.warn "Empty file: `#{filename}`"
         return
       end
-      stdid = doc.at("./publicationinfo/standard_id")&.text
-      return if stdid == "0"
+      return if doc.publicationinfo&.standard_id == "0"
 
-      fetcher = DataParser.new doc, self
-      bib = fetcher.parse
+      bib = IdamsParser.new(doc, self).parse
       if bib.docnumber.nil?
-        nt = doc&.at("./normtitle")&.text
-        Util.warn "PubID parse error. Normtitle: `#{nt}`, file: `#{filename}`"
+        Util.warn "PubID parse error. Normtitle: `#{doc.normtitle}`, file: `#{filename}`"
         return
       end
-      amsid = doc.at("./publicationinfo/amsid").text
+      amsid = doc.publicationinfo.amsid
       if backrefs.value?(bib.docidentifier[0].id) && /updates\.\d+/ !~ filename
         oamsid = backrefs.key bib.docidentifier[0].id
         Util.warn "Document exists ID: `#{bib.docidentifier[0].id}` AMSID: " \
              "`#{amsid}` source: `#{filename}`. Other AMSID: `#{oamsid}`"
-        if bib.docidentifier[0].id.include?(doc.at("./publicationinfo/stdnumber").text)
+        if bib.docidentifier[0].id.include?(doc.publicationinfo.stdnumber)
```

Comment:
This should be in the ieee-idams gem?

Reply:
I don't think so. The `backrefs` variable is a list of relation references which collected through all the documents and used later to create relations. The ieee-idams operates on a single document and shouldn't be involved in collection functionality.

Reply:
The update matches against the source filename path. It's not any kind of file content.